packages/common/src/api/tan-query/jupiter/useSwapTokens.ts

```diff
@@ -0,0 +1,375 @@
+        try {
+          quoteResult = await getJupiterQuoteByMint({
+            inputMint: inputMintUiAddress,
+            outputMint: outputMintUiAddress,
+            amountUi,
+            slippageBps,
+            swapMode: 'ExactIn',
+            onlyDirectRoutes: true
+          })
+        } catch (error: any) {
+          reportToSentry({
+            name: 'JupiterSwapQuoteError',
+            error,
+            feature: Feature.TanQuery,
+            additionalInfo: { params }
+          })
+          return {
+            status: SwapStatus.ERROR,
+            error: {
+              type: SwapErrorType.QUOTE_FAILED,
+              message: error?.message ?? 'Failed to get swap quote'
+            }
+          }
+        }
+
+        // ---------- 3. Prepare Transaction Instructions ----------
+        const inputTokenConfig =
+          USER_BANK_MANAGED_TOKENS[inputMintUiAddress.toUpperCase()]
+        const outputTokenConfig =
+          USER_BANK_MANAGED_TOKENS[outputMintUiAddress.toUpperCase()]
+
+        let sourceAtaForJupiter: PublicKey | undefined
+        let isSourceAtaTemporary = false
+
+        // --- 3a. Handle Input Token (if user bank managed) ---
+        const isInputUserBankManaged = !!(
+          inputTokenConfig &&
+          ethAddress &&
+          inputMintUiAddress.toUpperCase() !== 'SOL'
+        )
+        if (isInputUserBankManaged) {
+          try {
+            sourceAtaForJupiter = await addUserBankToAtaInstructions({
+              tokenInfo: inputTokenConfig!,
+              userPublicKey,
+              ethAddress: ethAddress!,
+              amountLamports: BigInt(quoteResult.inputAmount.amount),
+              sdk,
+              feePayer,
+              instructions
+            })
+            isSourceAtaTemporary = true
+          } catch (error: any) {
+            reportToSentry({
+              name: 'JupiterSwapInputPrepError',
+              error,
+              feature: Feature.TanQuery,
+              additionalInfo: { params }
+            })
+            return {
+              status: SwapStatus.ERROR,
+              error: {
+                type: SwapErrorType.BUILD_FAILED,
+                message: `Failed to prepare input token ${inputTokenConfig!.claimableTokenMint}: ${error.message}`
+              },
+              inputAmount: quoteResult.inputAmount,
+              outputAmount: quoteResult.outputAmount
+            }
+          }
+        }
+
+        // --- 3b. Determine Jupiter's Destination Token Account ---
+        let preferredJupiterDestination: string | undefined
+        let outputUserBankAddress: PublicKey | undefined
+
+        const isOutputUserBankManaged = !!(
+          outputTokenConfig &&
+          ethAddress &&
+          outputMintUiAddress.toUpperCase() !== 'SOL'
+        )
+        if (isOutputUserBankManaged) {
+          try {
+            outputUserBankAddress =
+              await sdk.services.claimableTokensClient.deriveUserBank({
```

Comment:
might be worth doing `getOrCreateUserBank` to ensure the destination exists?

alternatively, seems like the caller should specify destination and this fn shouldn't care about user banks as much as possible